target_chains/sui/sdk/js/src/client.ts

```diff
@@ -0,0 +1,298 @@
+    tx: TransactionBlock,
+    updates: number[][],
+    feedIds: HexString[]
+  ): Promise<ObjectId[]> {
+    const wormholePackageId = await this.getWormholePackageId();
+    const packageId = await this.getPythPackageId();
+
+    let priceUpdatesHotPotato;
+    if (updates.every((update) => this.isAccumulatorMsg(update))) {
+      if (updates.length > 1) {
+        throw new Error(
+          "SDK does not support sending multiple accumulator messages in a single transaction"
+        );
+      }
+      const vaa = this.extractVaaBytesFromAccumulatorMessage(updates[0]);
+      const verifiedVaas = await this.verifyVaas([vaa], tx);
+      [priceUpdatesHotPotato] = tx.moveCall({
+        target: `${packageId}::pyth::create_authenticated_price_infos_using_accumulator`,
+        arguments: [
+          tx.object(this.pythStateId),
+          tx.pure(updates[0]),
+          verifiedVaas[0],
+          tx.object(SUI_CLOCK_OBJECT_ID),
+        ],
+      });
+    } else if (updates.every((vaa) => !this.isAccumulatorMsg(vaa))) {
+      const verifiedVaas = await this.verifyVaas(updates, tx);
+      [priceUpdatesHotPotato] = tx.moveCall({
+        target: `${packageId}::pyth::create_price_infos_hot_potato`,
+        arguments: [
+          tx.object(this.pythStateId),
+          tx.makeMoveVec({
+            type: `${wormholePackageId}::vaa::VAA`,
+            objects: verifiedVaas,
+          }),
+          tx.object(SUI_CLOCK_OBJECT_ID),
+        ],
+      });
+    } else {
+      throw new Error("Can't mix accumulator and non-accumulator messages");
+    }
+
+    const priceInfoObjects: ObjectId[] = [];
+    for (const feedId of feedIds) {
+      const priceInfoObjectId = await this.getPriceFeedObjectId(feedId);
+      if (!priceInfoObjectId) {
+        throw new Error(
+          `Price feed ${feedId} not found, please create it first`
+        );
+      }
+      priceInfoObjects.push(priceInfoObjectId);
+      const coin = tx.splitCoins(tx.gas, [tx.pure(1)]);
+      [priceUpdatesHotPotato] = tx.moveCall({
+        target: `${packageId}::pyth::update_single_price_feed`,
+        arguments: [
+          tx.object(this.pythStateId),
+          priceUpdatesHotPotato,
+          tx.object(priceInfoObjectId),
+          coin,
+          tx.object(SUI_CLOCK_OBJECT_ID),
+        ],
+      });
+    }
+    tx.moveCall({
+      target: `${packageId}::hot_potato_vector::destroy`,
+      arguments: [priceUpdatesHotPotato],
+      typeArguments: [`${packageId}::price_info::PriceInfo`],
+    });
+    return priceInfoObjects;
+  }
+  async createPriceFeed(tx: TransactionBlock, updates: number[][]) {
+    const wormholePackageId = await this.getWormholePackageId();
+    const packageId = await this.getPythPackageId();
+    if (updates.every((update) => this.isAccumulatorMsg(update))) {
+      if (updates.length > 1) {
+        throw new Error(
+          "SDK does not support sending multiple accumulator messages in a single transaction"
+        );
```

Comment:
Why?

Reply:
Just for the sake of simplicity of the sdk and the fact that it's rare.